saleor/graphql/channel/mutations/base_channel_listing.py

```diff
@@ -66,8 +66,11 @@ def clean_channels(
         input_source="add_channels",
     ) -> Dict:
         add_channels = input.get(input_source, [])
-        add_channels_ids = [channel["channel_id"] for channel in add_channels]
+        add_channels_ids = [channel["channel_id"] for channel in add_channels
+                            if channel.get('remove_variants',[]) is not None]
```

Comment:
Providing default value for key that is not present in the dict is not necessary in this case
```suggestion
                            if channel.get('remove_variants')]
```

Reply:
In the current implementation if you want to add a new channel with variables like eg.
```
{
    "id": "UHJvZHVjdDoxNTI=",
    "input": {
      "removeChannels": null,
      "updateChannels": [
        {
          "channelId": "Q2hhbm5lbDoy",
          "isAvailableForPurchase": true,
          "isPublished": true,
          "removeVariants": null,
          "visibleInListings": true
        }
      ]
    }
  }
```
Then no channel was created.

In any situation when `removeVariants` is empty causes a situation when the whole object is deled from `updateChannels` list.

Reply:
Apologies I wrongly understood the issue the first time. As @fowczarek stated it shouldn't behave like that.
Fixed the mistake now. Please check the new commit.